### PR TITLE
v1.38 final merge

### PR DESCRIPTION
**Pull request by bernardkwok**
Update #460 
- Final merges before ADSK v1.38 release.
- Tests:
[v1.38_final_tests.pdf](https://github.com/autodesk-forks/MaterialX/files/6071842/v1.38_final_tests.pdf)

### CODE REVIEW

File: source/MaterialXCore/Document.cpp [hunk 1]
```diff
@@ -1327,28 +1334,7 @@ void Document::upgradeVersion()
...
-                            else if (input->getName() == INDEX_STRING)
-                            {
-                                input->setIsUniform(true);
-                            }
-                            else if (input->getName() == DEFAULT_STRING)
-                            {
-                                input->setIsUniform(true);
-                            }
-                        }
+                        input->setIsUniform(true);
```
Comment (bernardkwok) [inline source/MaterialXCore/Document.cpp:1337]: @niklasharrysson. We are merging this back now.

File: source/MaterialXRenderGlsl/TextureBaker.cpp [hunk 2]
```diff
@@ -444,10 +440,14 @@ BakedDocumentVec TextureBaker::createBakeDocuments(DocumentPtr doc, const FileSe
 
     DefaultColorManagementSystemPtr cms = DefaultColorManagementSystem::create(genContext.getShaderGenerator().getTarget());
     cms->loadLibrary(doc);
-    if (_codeSearchPath.isEmpty())
-        genContext.registerSourceCodeSearchPath(getDefaultSearchPath());
-    else
+    if (!_codeSearchPath.isEmpty())
```
Comment (bernardkwok) [inline source/MaterialXRenderGlsl/TextureBaker.cpp:443]: @ashwinbhat , @nicolassavva-autodesk : keeping this code alive. Do we still need this?

**Review by niklasharrysson — APPROVED** [review]
LGTM

File: source/MaterialXRenderGlsl/TextureBaker.cpp (hunk 2 above)
Comment (ashwinbhat) [reply source/MaterialXRenderGlsl/TextureBaker.cpp:443]: getDefaultSearchPath() works only for MtlxView I think

Comment (bernardkwok) [reply source/MaterialXRenderGlsl/TextureBaker.cpp:443]: Yes. The API call was removed and the code migrated to be inside MTLXView only. The checking for 
path / "libraries" was added but it's probably no better. Best to set an explicit path, thus I've left the API for this in while it's been removed from LucasFilm main.

**Pull request merged**